iml-gui/crate/src/components/stratagem/mod.rs

```diff
@@ -1,68 +1,455 @@
-use crate::{components::grafana_chart, generated::css_classes::C, GMsg};
-use iml_wire_types::Filesystem;
+use crate::{
+    components::{duration_picker, grafana_chart},
+    environment,
+    extensions::MergeAttrs,
+    generated::css_classes::C,
+    GMsg,
+};
+use futures::TryFutureExt;
+use iml_wire_types::{db::StratagemConfiguration, warp_drive::ArcCache, warp_drive::Locks, Filesystem, ToCompositeId};
 use seed::{prelude::*, *};
 use std::collections::HashMap;
+use std::sync::Arc;
 
+pub(crate) mod delete_stratagem_button;
+pub(crate) mod enable_stratagem_button;
 pub(crate) mod inode_table;
+pub(crate) mod scan_stratagem_button;
+pub(crate) mod scan_stratagem_modal;
+pub(crate) mod update_stratagem_button;
+
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct ActionResponse {
+    command: iml_wire_types::Command,
+}
+
+#[derive(Default)]
+pub struct TargetConfig {
+    pub interval: u64,
+    pub report_duration: Option<u64>,
+    pub purge_duration: Option<u64>,
+}
+
+#[derive(Debug, Default, Clone, serde::Serialize)]
+pub struct StratagemUpdate {
+    pub id: u32,
+    pub filesystem: u32,
+    pub interval: u64,
+    pub report_duration: Option<u64>,
+    pub purge_duration: Option<u64>,
+}
+
+#[derive(Debug, Default, Clone, serde::Serialize)]
+pub struct StratagemEnable {
+    pub filesystem: u32,
+    pub interval: u64,
+    pub report_duration: Option<u64>,
+    pub purge_duration: Option<u64>,
+}
+
+#[derive(Clone, Debug)]
+pub enum Command {
+    Update,
+    Delete,
+    Enable,
+}
+
+enum State {
+    Disabled,
```

Comment:
I think it is still can be manually scanned and all the tables and charts will make sense, no?

Reply:
The state will always begin as disabled when the filesystem page is loaded. An init function will send the `CheckStratagem` message. If appropriate, the state will be updated to enabled, in which case the chart and inode tables will be displayed.